### PR TITLE
Angular create user service

### LINKED ISSUE #51: Add User Service
User stories:

- #1

Interfaces:

- #46

### LINKED ISSUE #94: Add Message Logger

### LINKED ISSUE #95: Add Error Handler

### PR DESCRIPTION
**Pull request by dudaskobendeguz**
- closes #51 
- closes #94 
- closes #95

### CODE REVIEW

File: angular/src/app/app.module.ts [hunk 1]
```diff
@@ -6,11 +6,13 @@ import {AppRoutingModule} from './app-routing.module';
 import {AppComponent} from './app.component';
 import {HttpClientModule} from "@angular/common/http";
 import {InMemoryDataService} from "./services/in-memory-data.service";
+import { MessageLoggerComponent } from './components/message-logger/message-logger.component';
 
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
```
Comment (benedekhalaj) [inline angular/src/app/app.module.ts:14]: How thoughtful!

File: angular/src/app/components/message-logger/message-logger.component.html [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<div>
+  <h3>Messages: </h3>
+  <button type="button" (click)="messageLoggerService.clear()">Clear</button>
+  <ol>
+    <li *ngFor="let message of messageLoggerService.messages">
+      {{ message }}
```
Comment (benedekhalaj) [inline angular/src/app/components/message-logger/message-logger.component.html:6]: Nice component

File: angular/src/app/services/error-handler/error-handler.service.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    private messageLogger: MessageLoggerService
+  ) { }
+
+  handleError<T>(operation: string, result?: T): (error: any) => Observable<T> {
+    return (error: any): Observable<T> => {
+
+      // send the error to remote logging infrastructure
+      console.error(error);
+
+      // better job of transforming error for user consumption
```
Comment (benedekhalaj) [inline angular/src/app/services/error-handler/error-handler.service.ts:20]: thanks for comments again!

File: angular/src/app/services/user/user.service.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
```
Comment (benedekhalaj) [inline angular/src/app/services/user/user.service.ts:1]: We will use this as a template for future services.

**Review by benedekhalaj — APPROVED** [review]
**Thanks for the tutorial!** @dudaskobendeguz

**Review by Zsezsu — APPROVED** [review]
Nice job guys :)

**Pull request merged**